### PR TITLE
CON-3614

### PR DESCRIPTION
**Pull request by hkf-tech**

### CODE REVIEW

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Cohort/Delete.cshtml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+﻿@inject SFA.DAS.ProviderUrlHelper.ILinkGenerator LinkGenerator
+@using SFA.DAS.ProviderCommitments.Web.Extensions
+@model SFA.DAS.ProviderCommitments.Web.Models.Cohort.DeleteCohortViewModel
+
+
+@{
+    ViewBag.Title = "Confirm Delete Cohort";
+    ViewBag.PageId = "confirm-delete-cohort";
```
Comment (VasanthaKasirajan3008) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Cohort/Delete.cshtml:8]: can we add GaData.Vpv and GaData.Org

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Cohort/Delete.cshtml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                                    <dd class="govuk-summary-list__value">@Model.ApprenticeshipTrainingProgrammes.First()</dd>
+                                }
+                            </div>
+                        </dl>
+                    </div>
+                </div>
+                <form method="post">
+                    <fieldset class="govuk-fieldset">
+                        @Html.AntiForgeryToken()
+                        @Html.HiddenFor(x => x.ProviderId)
```
Comment (VasanthaKasirajan3008) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Cohort/Delete.cshtml:51]: not sure .. ProviderId in hidden field

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DeleteCohortRequestViewModelMapper.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using SFA.DAS.CommitmentsV2.Api.Client;
+using SFA.DAS.CommitmentsV2.Shared.Interfaces;
+using SFA.DAS.ProviderCommitments.Web.Models.Apprentice;
+using SFA.DAS.ProviderCommitments.Web.Models.Cohort;
+
```
Comment (VasanthaKasirajan3008) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DeleteCohortRequestViewModelMapper.cs:8]: remove unused namespace

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/CohortController.cs [hunk 4]
```diff
@@ -138,5 +139,32 @@ public async Task<IActionResult> ConfirmEmployer(ConfirmEmployerViewModel viewMo
...
+
+        [HttpPost]
+        [Route("{cohortReference}/details/delete")]
+        [DasAuthorize(ProviderFeature.ProviderCreateCohortV2)]
+        [Authorize(Policy = nameof(PolicyNames.HasContributorOrAbovePermission))]
+        public async Task<IActionResult> Delete([FromServices] IAuthenticationService authenticationService, DeleteCohortViewModel viewModel)
+        {
+            if (viewModel.Confirm.Value)
+            {
+                CommitmentsV2.Types.UserInfo userInfo = authenticationService.UserInfo;
```
Comment (VasanthaKasirajan3008) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/CohortController.cs:162]: why do we need  to set this explicitly here ?

Comment (hkf-tech) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/CohortController.cs:162]: If we don't set this, the history table isn't populated correctly - i.e. the user who made the change.

**Review by VasanthaKasirajan3008 — APPROVED** [review]

**Pull request merged**